docs/advanced_installation.md

```diff
@@ -0,0 +1,176 @@
+# Advanced Installation
+
+This installation is for advanced users that want to contribute to the project
+and/or add new resources or modify existing resources.
+
+This will require some
+additional dependencies and a knowledge of
+[Kustomize](https://kubernetes-sigs.github.io/kustomize/guides/).
+
+## Requirements
+
+### Files and Services
+
+You need to have the following:
+
+* access to a Kubernetes cluster running at least version `1.11.3` and
+[kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/)
+properly set up
+* a Project in [Google Cloud](https://console.cloud.google.com/) with
+[Service Directory](https://cloud.google.com/service-directory) enabled
+* a [Google Cloud Service Account](https://cloud.google.com/iam/docs/service-accounts)
+
+### Programs
+
+Please make sure you have the following programs installed:
+
+* [Kubectl 1.11.3+](https://kubernetes.io/docs/tasks/tools/install-kubectl/):
+  * *Unix/Linux* users with
+  [Snap](https://snapcraft.io/docs/installing-snapd):
+
+    ```bash
+    snap install kubectl --classic
+    ```
+
+  * *MacOs* users with [HomeBrew](https://brew.sh/):
+
+    ```bash
+    brew install kubectl
+    ```
+
+  * *Windows* users: follow
+  [this section](https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-windows)
+  of the documentation
+* [Make](https://www.gnu.org/software/make/), which will automate some steps:
+  * *Unix/Linux and MacOS*: already pre-installed.
+  * *Windows* users: download the binaries from
+  [this page](http://gnuwin32.sourceforge.net/packages/make.htm).
+* [Golang 1.13+](https://golang.org/doc/install) to build the project. Follow
+the link to learn how to install it for any system.
+* [Docker 17.03+](https://www.docker.com/get-started) for building and pushing
+the operator's container images.
+  * *Unix/Linux* users with
+  [Snap](https://snapcraft.io/docs/installing-snapd):
+
+  ```bash
+  sudo snap install docker
+  ```
+
+  * *MacOs* users:
+  [Docker Desktop for Mac](https://hub.docker.com/editions/community/docker-ce-desktop-mac/)
+  * *Windows* users:
+  [Docker Desktop for Windows](https://hub.docker.com/editions/community/docker-ce-desktop-windows/)
+
+* [Kubebuilder](https://github.com/kubernetes-sigs/kubebuilder#installation):
+  * *MacOs* users with [HomeBrew](https://brew.sh/):
+
+  ```bash
+  brew install kubebuilder
+  ```
+
+  * *Other systems*: follow
+  [this section](https://book.kubebuilder.io/quick-start.html#installation)
+  on the documentation.
+
+### Your files
+
+You will need to place files depending on what kind of files you want to modify.
```

Comment:
Not sure I understand this sentence. In general the "Your files" section it's a bit confusing to me (it may be just me)

Reply:
That's for adding resources for advanced users. For example you may add new Roles, Deployments, Services etc. Depending on the kind of the resource you need to place the file on the appropriate folder, so roles go to rbac folder and deployments/services go to manager and so on (but mostly users will want to place them in these two folders).

People that go this way already know about kubernetes & operators for the most part, so that's why I did not elaborate it too much.

Reply:
Shall we refer to these as "custom files" then? Is that name accurate?

Reply:
I think that custom files would mostly remind of crds. I'll see how I can reword this

Reply:
I rephrased the whole section and also specified to not follow the guide if you're not touch Kubernetes resources.
I am actually thinking about removing those files and the dependency from Kustomize entirely. I will have to think about this a bit.